libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -847,6 +853,73 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         )
     }
 
+    /***
+     * Sets routes to preview. Triggers an update in [RoutesPreviewObserver].
+     *
+     * If [routes] isn't empty, the route with [primaryRouteIndex] is considered as primary, the others as alternatives.
+     * To cleanup routes preview state pass an empty list as [routes].
+     *
+     * Use [RoutesPreview.routesList] to start Active Guidance after route's preview:
+     * ```
+     * mapboxNavigation.getRoutesPreview()?.routesList?.let{ routesList ->
+     *     mapboxNavigation.setNavigationRoutes(routesList)
+     * }
+     * ```
+     * Routes preview state is controlled by the SDK's user. If you want to stop routes preview when you start active guidance, do it manually:
+     * ```
+     * mapboxNavigation.setRoutesPreview(emptyList())
+     * ```
+     *
+     * @param routes to preview
+     * @param primaryRouteIndex index of primary route from [routes]
+     */
+    @ExperimentalPreviewMapboxNavigationAPI
+    @JvmOverloads
+    fun setRoutesPreview(routes: List<NavigationRoute>, primaryRouteIndex: Int = 0) {
+        routesPreviewController.previewNavigationRoutes(routes, primaryRouteIndex)
+    }
+
+    /***
+     * Changes primary route for current preview state without changing order of [RoutesPreview.originalRoutesList].
+     * Order is important for a case when routes are displayed as a list on UI, the list shouldn't change order when a user choose different primary route.
+     *
+     * In case [changeRoutesPreviewPrimaryRoute] is called while the the other set of routes are being processed, the current state with a new routes will be reapplied after the current processing.
+     *
+     * @param newPrimaryRoute is a new primary route
+     * @throws [IllegalArgumentException] if [newPrimaryRoute] isn't found in the previewed routes list
+     */
+    @ExperimentalPreviewMapboxNavigationAPI
+    @Throws(IllegalArgumentException::class)
+    fun changeRoutesPreviewPrimaryRoute(newPrimaryRoute: NavigationRoute) {
```

Comment:
What if we call it `replaceRoutesPreviewPrimaryRoute`? "replace" is used in String and Collection methods to indicate that you replace an existing value.
it's not perfect but slightly better I think.

Reply:
Well, yes, but here the first argument is kind of in the method name ("RoutesPreviewPrimaryRoute"). But OK, agreed that one may think that the old value should be passed here. What about `replaceRoutesPreviewPrimaryRouteWith`?

Reply:
TBH "change" seems less misleading for me than "replace"

Reply:
OK, let's leave as-is.

Reply:
making a list of options is helpful sometimes

- changeRoutesPreviewPrimaryRoute
- replaceRoutesPreviewPrimaryRoute
- setRoutesPreviewPrimaryRoute
- selectRoutesPreviewRoute
- setRoutesPreviewRoute

I typically prefer the one easiest to read and find, so i'll throw in a vote for `setRoutesPreviewRoute`